front/src/components/boxs/device-in-room/DevicesInRoomsBox.jsx

```diff
@@ -1,77 +1,94 @@
 import { Component } from 'preact';
 import { connect } from 'unistore/preact';
+import get from 'get-value';
+import cx from 'classnames';
+
 import actions from '../../../actions/dashboard/boxes/devicesInRoom';
 import { RequestStatus, DASHBOARD_BOX_STATUS_KEY, DASHBOARD_BOX_DATA_KEY } from '../../../utils/consts';
-import { WEBSOCKET_MESSAGE_TYPES } from '../../../../../server/utils/constants';
-import get from 'get-value';
+import {
+  DEVICE_FEATURE_CATEGORIES,
+  DEVICE_FEATURE_TYPES,
+  WEBSOCKET_MESSAGE_TYPES
+} from '../../../../../server/utils/constants';
 
 import DeviceRow from './DeviceRow';
 
-const minHeight = {
-  minHeight: '6rem'
+const enableSwithAll = feature => {
+  return (
+    feature.category === DEVICE_FEATURE_CATEGORIES.LIGHT &&
+    feature.type === DEVICE_FEATURE_TYPES.LIGHT.BINARY &&
+    feature.read_only === false
+  );
 };
 
 const changeAllLightsStatusRoom = (props, boxData) => () => {
   const newStatus = boxData.roomLightStatus === 1 ? 0 : 1;
   props.changeAllLightsStatusRoom(props.x, props.y, newStatus);
 };
 
-const RoomCard = ({ children, ...props }) => {
+const DeviceInRoomCard = ({ children, ...props }) => {
+  const { roomName, boxData, loading, devices = [], box = {} } = props;
+  const { device_features: featureSelectors = [] } = box;
+
+  const hasBinaryLightDeviceFeature =
+    devices
+      .flatMap(device => device.features)
+      .filter(feature => featureSelectors.indexOf(feature.selector) !== -1)
+      .filter(feature => enableSwithAll(feature)).length > 0;
```

Comment:
Looks very heavy to me to do this on every render, it's 3 loops on the device array, and we just need to know if we find one, why not use a .find instead of a doing many filtering ? 

Find stops when it find one element that match the function, whereas filter will continue until the end of the array

Reply:
Now finding first matching feature in first matching device.
Thanks.